services/ui-src/src/components/fields/NumberField.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { useState } from "react";
+import { useFormContext } from "react-hook-form";
+// components
+import { Box } from "@chakra-ui/react";
+import { TextField } from "./TextField";
+// utils
+import { isCustomMask, makeMediaQueryClasses, maskValue } from "utils";
+import { InputChangeEvent, AnyObject } from "types";
+
+export const NumberField = ({
+  name,
+  label,
+  placeholder,
+  mask,
+  sxOverride,
+  ...props
+}: Props) => {
+  const mqClasses = makeMediaQueryClasses();
+
+  const [displayValue, setDisplayValue] = useState(props?.hydrate || "");
+
+  // get the form context
+  const form = useFormContext();
+
+  const onBlurHandler = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const eventValue = isCustomMask(mask)
+      ? maskValue(e.target.value, mask)
+      : e.target.value;
+    setDisplayValue(eventValue);
+    form.setValue(name, eventValue, { shouldValidate: true });
+  };
+
+  // update form data
+  const onChangeHandler = async (e: InputChangeEvent) => {
+    const { name, value } = e.target;
+    setDisplayValue(value);
+    form.setValue(name, value, { shouldValidate: true });
+  };
+
+  /*
+   * Check if its not a custom mask and if the mask is defined as "currency"
+   * or another CMSDS provided mask
+   * If its not, we dont want to use the mask prop so return undefined.
+   */
+  const isCmsdsProvidedMask = mask && !isCustomMask(mask) ? mask : undefined;
+
+  // const maskToUse = customMaskMap[mask] || mask;
```

Comment:
Nit: Since this is actually returning the mask now, we could rename it. 
```suggestion
  const cmsdsPassedMask = mask && !isCustomMask(mask) ? mask : undefined;
```